src/overlays/actors/ovl_En_Rl/z_en_rl.c

```diff
@@ -9,81 +16,377 @@ void EnRl_Destroy(Actor* thisx, GlobalContext* globalCtx);
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE7CE8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE7D40.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE7D94.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/EnRl_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/EnRl_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE7FD0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE7FDC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/EnRl_Draw.s")
+void func_80AE7798(EnRl* this, GlobalContext* globalCtx);
+void func_80AE77B8(EnRl* this, GlobalContext* globalCtx);
+void func_80AE77F8(EnRl* this, GlobalContext* globalCtx);
+void func_80AE7838(EnRl* this, GlobalContext* globalCtx);
+void func_80AE7C64(EnRl* this, GlobalContext* globalCtx);
+void func_80AE7C94(EnRl* this, GlobalContext* globalCtx);
+void func_80AE7CE8(EnRl* this, GlobalContext* globalCtx);
+void func_80AE7D40(EnRl* this, GlobalContext* globalCtx);
+void func_80AE7FD0(EnRl* this, GlobalContext* globalCtx);
+void func_80AE7FDC(EnRl* this, GlobalContext* globalCtx);
+void func_80AE7D94(EnRl* this, GlobalContext* globalCtx);
+
+UNK_PTR D_80AE81A0[] = { 0x06003620, 0x06003960, 0x06003B60 };
+
+extern SkeletonHeader D_06007B38;
+extern AnimationHeader D_06000A3C;
+extern AnimationHeader D_06000830;
+extern AnimationHeader D_0600040C;
+
+void EnRl_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnRl* this = THIS;
+
+    SkelAnime_Free(&this->skelAnime, globalCtx);
+}
+
+void func_80AE72D0(EnRl* this) {
+    s32 pad[3];
+    s16* timer = &this->timer;
+    s16* eyeTextureIndex = &this->eyeTextureIndex;
+
+    if (DECR(*timer) == 0) {
+        *timer = Math_Rand_S16Offset(60, 60);
+    }
+
+    *eyeTextureIndex = *timer;
+    if (*eyeTextureIndex > 2) {
+        *eyeTextureIndex = 0;
+    }
+}
+
+void func_80AE7358(EnRl* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000A3C, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_06000A3C.genericHeader),
+                         0, 0.0f);
+    this->action = 4;
+    this->drawConfig = 0;
+    this->alpha = 0;
+    this->lightBallSpawned = 0;
+    this->actor.shape.unk_14 = 0;
+    this->unk_19C = 0.0f;
+}
+
+void func_80AE73D8(EnRl* this, GlobalContext* globalCtx) {
+    static s32 D_80AE81AC = 0;
+
+    if (globalCtx->csCtx.state == 0) {
+        if (D_80AE81AC) {
+            if (this->actor.params == 2) {
+                func_80AE7358(this);
+            }
+            D_80AE81AC = 0;
+        }
+    } else if (!D_80AE81AC) {
+        D_80AE81AC = 1;
+    }
+}
+
+EnRlActionFunc sActionFuncs[] = { func_80AE7798, func_80AE77B8, func_80AE77F8, func_80AE7838,
+                                  func_80AE7C64, func_80AE7C94, func_80AE7CE8, func_80AE7D40 };
```

Comment:
We leave trailing commas in cases like these to help with the vertical formatting (from clang-format). Make sure to run clang-format after addressing these comments.
```suggestion
                                  func_80AE7C64, func_80AE7C94, func_80AE7CE8, func_80AE7D40, };
```